### PR TITLE
Runtime Plugins

### PR DESCRIPTION
**Pull request by jgbernalp**
# Description

This PR is a PoC for the new runtime plugins architecture. The core plugins involved in e2e tests were extracted as a runtime plugin living in: https://github.com/perses/perses-plugins

# Screenshots

# Checklist

- [x] Pull request has a descriptive title and context useful to a reviewer.
- [x] Pull request title follows the `[<catalog_entry>] <commit message>` naming convention using one of the
  following `catalog_entry` values: `FEATURE`, `ENHANCEMENT`, `BUGFIX`, `BREAKINGCHANGE`, `DOC`,`IGNORE`.
- [x] All commits have [DCO signoffs](https://github.com/probot/dco#how-it-works).

## UI Changes

Use a Dynamic panel to load plugins at runtime

## How to test
- run perses ui
```bash
cd ui && npm run start
```
- run perses backend for development, this will install the default plugins from https://github.com/perses/perses-plugins
```bash
./scripts/api_backend_dev.sh
```
- check Perses UI. **Note** that only the following plugins will be available:
  - BarChart
  - GaugeChart
  - Markdown
  - StatChart
  - TimeSeriesChart

### CODE REVIEW

File: ui/app/package.json [hunk 1]
```diff
@@ -4,12 +4,12 @@
   "private": true,
   "scripts": {
     "clean": "rimraf dist/",
-    "start": "webpack serve --config webpack.dev.ts",
-    "build": "webpack --config webpack.prod.ts",
+    "start": "NODE_ENV=development rspack serve",
```
Comment (AntoineThebaud) [inline ui/app/package.json:7]:
```suggestion
    "start": "cross-env NODE_ENV=development rspack serve",
```

**Review by Nexucis — COMMENTED** [review]
few thoughts around the backend

File: docs/examples/config.docker.yaml [hunk 2]
```diff
@@ -9,3 +9,6 @@ schemas:
   datasources_path: "/etc/perses/cue/schemas/datasources"
   variables_path: "/etc/perses/cue/schemas/variables"
   interval: "6h"
+
+frontend:
+  plugins_path: "/etc/perses/plugins"
```
[the remark this answers is not in the document]
Comment (jgbernalp) [reply docs/examples/config.docker.yaml:14]: agreed as they are in both backend and frontend, will adjust.

File: distroless-debug.Dockerfile [hunk 3]
```diff
@@ -13,6 +13,7 @@ COPY --chown=nobody:nobody percli                            /bin/percli
 COPY --chown=nobody:nobody LICENSE                           /LICENSE
 COPY --chown=nobody:nobody cue/schemas/                      /etc/perses/cue/schemas/
 COPY --chown=nobody:nobody cue.mod/                          /etc/perses/cue.mod/
+COPY --chown=nobody:nobody plugins/                          /etc/perses/plugins/
```
[the remark this answers is not in the document]
Comment (Nexucis) [reply distroless-debug.Dockerfile:16]: I am not sure either. We can start with the default image containing all core plugin and see if someone would like the empty image

**Comment by Nexucis** [comment]
I think regardless the issues with Happo, once the last fixes on the backend are done, we can merge. Like that we can iterate base on your work @jgbernalp.

Issues with Happo can be fixed in a later PR, it won't change the architecture I believe

File: pkg/model/api/config/config_test.go [hunk 4]
```diff
@@ -352,6 +362,9 @@ frontend:
       dashboard: "DuplicatePanels"
     - project: "Unknown"
       dashboard: "Dashboard"
+
+plugins:
+  path: "custom/plugins"
```
Comment (Nexucis) [inline pkg/model/api/config/config_test.go:367]: the tests are not passing, because you put the plugins config in the middle of the frontend config :)

Comment (jgbernalp) [reply pkg/model/api/config/config_test.go]: silly me, thnx for the hint. It should be ready now.

**Comment by Nexucis** [comment]
nice ! Thank you for this awesome work @jgbernalp !!

**Review by Nexucis — APPROVED** [review]

**Pull request merged**